pico_limbo/src/handlers/configuration.rs

```diff
@@ -224,6 +225,14 @@ pub fn send_play_packets(
     let packet = UpdateTimePacket::new(ticks, ticks, !lock_time);
     client_state.queue_packet(PacketRegistry::UpdateTime(packet));
 
+    let tablist = server_state.tablist();
+    if let Some(header) = tablist.header.as_ref()
+        && let Some(footer) = tablist.footer.as_ref()
+    {
+        let packet = TabListPacket::new(header, footer);
+        client_state.queue_packet(PacketRegistry::TabList(packet));
+    }
```

Comment:
If you use a fat enum, this will make this code easier to read:

```rust
  match server_state.tablist() {
    HeaderAndFooter { header, footer } => TabListPacket::new(header, footer),
    Header { header } => TabListPacket::new(header, None),
    Footer { footer } => TabListPacket::new(None, footer),
    None => {},
  }
```

Maybe you can even create other constructors for TabListPacket that only accept one or the other.
This packet should not be sent if the tab list is None